api/krusty/remoteload_test.go

```diff
@@ -74,6 +77,31 @@ spec:
 	assert.NoError(t, fSys.RemoveAll(tmpDir.String()))
 }
 
+func TestRemoteResourceWithHTTPError(t *testing.T) {
+	fSys := filesys.MakeFsOnDisk()
+	b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+	tmpDir, err := filesys.NewTmpConfirmedDir()
+	assert.NoError(t, err)
+
+	url404 := "https://github.com/thisisa404.yaml"
+	kusto := filepath.Join(tmpDir.String(), "kustomization.yaml")
+	assert.NoError(t, fSys.WriteFile(kusto, []byte(fmt.Sprintf(`
+resources:
+- %s
+`, url404))))
+
+	_, err = b.Run(fSys, tmpDir.String())
+	if utils.IsErrTimeout(err) {
+		// Don't fail on timeouts.
+		t.SkipNow()
+	}
+
+	httpErr := fmt.Errorf("%w: status code %d (%s)", loader.ErrorHTTP, 404, http.StatusText(404))
+	accuFromErr := fmt.Errorf("accumulating resources from '%s': %w", url404, httpErr)
+	expectedErr := fmt.Errorf("accumulating resources: %w", accuFromErr)
```

Comment:
So the error we will get is `accumulating resources: `accumulating resources from 'https://github.com/thisisa404.yaml': HTTP error: status code 404 (not found)`.  That's a pretty good description, probably much more helpful than whatever we were outputting before.